### DISCUSSION TITLE
How to run example directory?

### DISCUSSION THREAD

**Discussion by SHND** [gvergnaud/ts-pattern#127] [Q&A] [answered]
I followed the steps below, but couldn't run the exmaple directory in this repo.

```
cd examples
npm install
tsc
```

But this gives me an error. What are the steps to run the example?

**Comment by gvergnaud** [comment, marked as answer]
hmm I think this should work, what error do you get?

The easiest is probably to copy/paste the content of `one-file-example.ts` to a typescript playground: [Link](https://www.typescriptlang.org/play?#code/PQKhCgAIUhifIHkB2BTSAzAlgG3QFQGUBaABQEMAXS1AJ2UgBNUBbAewDooJpJ8ALLAGdIAd1w4mrNsiGVaVVAC5ukYpAASbUZEptIAVyHoADlRr1ILKgGNByAOaRyyRmNyMb5WoyGr1Wjp6kKgAHvLkNpTOkABu5DgG6Bi0bCy6-OhYyCYG0QAiVOSQhPIGUQa06EbZTgAGpBzGeFF1-prauvrWlHaQxrF0CZBexiI1jpANHAbIWDJtvAGdwXiDCg7oYfzkRpRYg2hCInaoNgDWtV1W5OfoQpVb6wCeI+TGkMKQO654jO2Ba5mah0Bg9PoyfqodbDeKJVAiKiQGQ2apCK6UAwmPB+JYdIL6OFYRiKZwMWbnZDaBgAQVIAElIFUhCYZB8JvVhABZWz2Bx1MluYEWWSQHBYO6qSBTRreBTPOoAGhlHDYJn2MgSSpCvS40GA4HAWBYrNo0QA3p8hDzenzleD+MrSJAAL6YVLpADklCExGFoM9AG5DaAwGHwxHI1HozGI6oAKofUboACMSm+LkY4smjGENloxuyijcs3mDEozxMCOg3FjdfrDbDBvAFarkAASgjWbJ0ABeKCQAA+kEtreUkE9sWJqDYQaYRXTlowbFoPXTnpNABZPUOJ6JUAAjFhzoS0GzpuQFyZul0D4ejyvjz3G8ibOckyjkRchcKoWRl9cHCwDAd2HT0ACsTAcUCJxMRwTzPC95CuG87xHXRH3XGhwnfBd+mQxxA10V8hG-ZByBYcdL1qIjiXTZADBYA86FdABtABdV1g3AGw2WiMIKOxVAADFZiiMsU0gXtIAACmyXJKHTTsWTZVAAEokKvJxewAPgHB05JyPI1IHaVgGASAUw4SAACF3iwGxIH9SxxEofhPmQDA6D-VF3DcmJxBwTxvH+aVpQ4Vz+Bkh8q3XKdmFnZUPy-dDl1XKh1y3Hcb2VGT4pnNSpJ0uJp04ZKmjPEywsgczIAAJmsjkoRaShTMgCKsDcmS2ulGKnxfN8krwy0whof8ZEA4DPWVU9z0gRpmjOSgZMKnKetk2bCt0qYAB5jScWbewAEnNWa3QSShewAIgEL4vmKICMAAQiumqdMWMKqrC2qAGZGvRSZGlLGQ2o6rr1vNdbpTHdcBtQaaofnT9v1Gv90Qm+aZjmGQZIgqDptg+C1JmxDMcWqIVtdRV1pdanqulGTNqK3b9v6M9jtOs9zpwS6bsEER7sgSCnBXGI4IcF63o+6UvrMizN2s8n9kmBJJE-JxyMokRIsxuVyFeTMydQFrQci7r6fQmGJ2wygCfV0jddoeVosgTXxwW42lspl1Vrp+mZPVgA5CiES24q6htvz3Pt9MTqDkOhA4cC2GyXHlU9NSXWlyBZZqiyAFZrIPezHO2XY5AOP8EROTILlqNrasAArJAHg-pg2ARZBvQwqtiFOC4nobiyAAEfWIMIqyiMenZXb8rcnUqCeS780rXPdD2PV1N+ETvol49ZUFCsKODLvZK5W7jQ0bK+owTJN3nQOr0xYAweawJzvAokQDwUZA7CuHhr6AIjM2MckBEx0HwI+KSE4D6dRXDBeeqB9y0CDOAWq8Zf5pEosgaIno-wWBMAWYwO5gjGC2KEHYp9BgjFrpcSY2RnDiRBqAxAtAHCkBwC4aBnpi7ohsAgwhs5dyekIawLAjE5y1TAvgughDhDw24rxWQ-FQiCTwKJX+GpkB1WgTJFcDh0ysPYZw5AyojB0HTOA2gkCqxhwHLVKykAADCtCrgDCGJIP8jFtadWjliHE+leQyVYvosxxhaCcXeCMPiucwZRVYjwkuBNGgAH12K5TDjKPA99DBQQUMwa4swcBsAuBkL4XkqCPCenUXOtUGpTDiStAUXhyy3HuNCDx78QT0ETpAek0RAqSAdJ8DAyI0DIlGW5Vgbc9RHzNutBJgiCaIOQZ6dJCyRFVBYOI486cpxILoGsv21VKbbTqAATTYJURhvFZjREYO3IQO8MzUKrKuYQ6NRTBHMWUkQFTMRVGzrEs2izUjLNgXoFB6TZKZLqBoY2xTqnAt8cEzZYiJHpwhfA6Fpzw6XIMM4KoyJrkYHILEFcnV0A2D2GkOgSLB6QD+tEk0eBQg-koRXahfc6EOFBifTlqBz4hiASKuMvBwFvA+D9dMAA1BIxIqBljJJAOkjJmTdg+AA0V2rmxKLkIYcJFBukMGkpDaUbt0wLQIg4P2DjrJqi0cMQRbz9gIgHK+d2qp1RlgSDJRoDEmJ0GJgOIQJSsAJAADLZHOA7RoDqfU4Gim1SgrkLCWqaNa45Hk5CvgUCwdN1FHB+x9tTF0ii+JOTYHII1IpoFmt0J1PABbM0DiUWNRSZMW3SnFHcJxVycHpoDcxWgftdhuRngaugNbQS2osvUvWzxcTMuwT6dNC6k3VTHfwCd5jaDTvoFmtt+Dm1aWLcGsthpQGkCrdEaSjRsheVoDtMcbBRmsmreYUEOkL4WQAKI4GMKITIRKGFTOicwFQtU6heVtDJDg8G1IcCmcgPK8qPyoGUhq9SbQeIVqJOhzDqldHqtUumCkVJRDIA0vNG9u4GI4EkNtetv1rL4cVZMcj1IPIKSjlaG0f9HADmArJbkvJagyXfZQfdpimRdlUmpQq9a5bShIz2K0HlH1VDcFEmUN66hcAZRg3iLAV0ZAROgbM1ZhPPCuWIFw0RgjyFeE1OoqnjB1BUNVWqerd4hXwI2vsUwTpudQEhgLboDyvCHsFuTPYOBbpXBwN2WdgxeYsj55dMjaAiGkiFjgxmV2J2sCYGSMkbCZJsPFvI27aBJZDoh5OqdllqVS2FKoAKGAhda7eAc7XKgMHozgYMZagA)


[This section of the README](https://github.com/gvergnaud/ts-pattern#code-sandbox-examples) also contains links to codesandboxes with ts-pattern examples

**Comment by SHND** [comment]
Adding `lib: "es2015"` fixed the compile issues.

I opened a PR here if this is a right fix: https://github.com/gvergnaud/ts-pattern/pull/128

_tsc Version 4.8.4
node Version v16.13.1
npm Version 8.1.2_

The 2 top errors were:

```
node_modules/ts-pattern/dist/types/BuildMany.d.ts:18:110 - error TS2583: Cannot find name 'Set'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2015' or later.

18 ] : never : data extends readonly (infer a)[] ? Update<a, value, Cast<tail, PropertyKey[]>>[] : data extends Set<infer a> ? Set<Update<a, value, Cast<tail, PropertyKey[]>>> : data extends Map<infer k, infer v> ? Map<k, Update<v, value, Cast<tail, PropertyKey[]>>> : Compute<Omit<data, Cast<head, PropertyKey>> & {
                                                                                                                ~~~

node_modules/ts-pattern/dist/types/BuildMany.d.ts:18:125 - error TS2583: Cannot find name 'Set'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2015' or later.

18 ] : never : data extends readonly (infer a)[] ? Update<a, value, Cast<tail, PropertyKey[]>>[] : data extends Set<infer a> ? Set<Update<a, value, Cast<tail, PropertyKey[]>>> : data extends Map<infer k, infer v> ? Map<k, Update<v, value, Cast<tail, PropertyKey[]>>> : Compute<Omit<data, Cast<head, PropertyKey>> & {
                                                                                                                               ~~~
...
```
